src/ApiClient.js

```diff
@@ -90,6 +90,10 @@ class ApiClient {
         // else send back the parsed error
         .then(parsedErr => { throw parsedErr })
     }
+    // fix of "Unexpected end of JSON input" when HTTP response is 204 No Content
+    if (response.status === 204) {
+      return { json: () => () => null }
+    }
```

Comment:
I'm assuming you have a good test env set up for this change. I think also that response.json() should return a string (or null), not another function, i.e.:

```
return { json: () => null }
```

Could you also do me a favour and see if this code works instead?